test/spec/downloadTests.ts

```diff
@@ -26,16 +31,54 @@ if (fs.existsSync(outputDir)) {
 
 fs.mkdirSync(outputDir, {recursive: true});
 
-for (const test of SPEC_TEST_TO_DOWNLOAD) {
-  const url = `${specTestsRepoUrl}/releases/download/${specVersion}/${test}.tar.gz`;
-  console.log(`Downloading ${url} to ${outputDir}`);
-  shell(`wget -c ${url} -O - | tar -xz --directory ${outputDir}`);
-  console.log(`Downloaded ${url} to ${outputDir}`);
-}
+const urls = SPEC_TEST_TO_DOWNLOAD.map((test) => `${specTestsRepoUrl}/releases/download/${specVersion}/${test}.tar.gz`);
 
-// If successful
-fs.writeFileSync(versionFile, specVersion);
+downloadAndExtract(urls, outputDir)
+  .then(() => {
+    console.log("Downloads and extractions complete.");
+    fs.writeFileSync(versionFile, specVersion);
+  })
+  .catch((error) => {
+    console.error(`Error downloading test files: ${error}`);
+    process.exit(1);
+  });
 
 function shell(cmd: string): string {
-  return execSync(cmd, {encoding: "utf8"}).trim();
+  try {
+    return execSync(cmd, {encoding: "utf8"}).trim();
+  } catch (error) {
+    console.error(`Error executing shell command: ${cmd}`);
+    throw error;
+  }
+}
+
+async function downloadAndExtract(urls: string[], outputDir: string): Promise<void> {
```

Comment:
Look on the caller of `downloadAndExtract` not inside the function
```suggestion
async function downloadAndExtract(url: string, outPath: string): Promise<void> {
```

Reply:
Assuming you mean loop, any particular reason? this function was intended to download all the test files